pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -353,12 +360,23 @@ def on_save_checkpoint(
     def on_load_checkpoint(
         self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", callback_state: Dict[str, Any]
     ) -> None:
-        self.best_model_score = callback_state["best_model_score"]
-        self.best_model_path = callback_state["best_model_path"]
-        self.best_k_models = callback_state.get("best_k_models", self.best_k_models)
-        self.kth_best_model_path = callback_state.get("kth_best_model_path", self.kth_best_model_path)
-        self.kth_value = callback_state.get("kth_value", self.kth_value)
+        dirpath_from_ckpt = callback_state.get("dirpath", self.dirpath)
+
+        if self.dirpath == dirpath_from_ckpt:
+            self.best_model_score = callback_state["best_model_score"]
+            self.kth_best_model_path = callback_state.get("kth_best_model_path", self.kth_best_model_path)
+            self.kth_value = callback_state.get("kth_value", self.kth_value)
+            self.best_k_models = callback_state.get("best_k_models", self.best_k_models)
+        else:
+            # If dirpath has changed, raise a warning and only track last_model_path
+            # and best_model_path
+            warnings.warn(
+                f"The dirpath was changed from {self.dirpath} to {dirpath_from_ckpt},"
```

Comment:
```suggestion
                f"The dirpath has changed from {dirpath_from_ckpt!r} to {self.dirpath!r},"
```
from old_path to new_path, right?

Reply:
```suggestion
                f"The dirpath has changed from {dirpath_from_ckpt} to {self.dirpath},"
```

Reply:
> from old_path to new_path, right?

Right, good catch.